core/src/main/scala/besom/types.scala

```diff
@@ -68,20 +116,31 @@ object types:
 
   end ProviderType
 
+  /** A special [[ResourceType]] that identifies a Pulumi provider function.
+    */
   opaque type FunctionToken <: PulumiToken = String
 
+  /** See [[FunctionToken]] type for more information.
+    */
   object FunctionToken:
+    /** Parse a function token [[String]] into a [[FunctionToken]].
+      * @param s
+      *   a function token string to parse
+      * @return
+      *   a [[FunctionToken]] if the string is valid, otherwise an compile time error occurs
+      */
     // validate that function token contains two colons between three identifiers, see @ResourceType
     inline def from(s: String): FunctionToken =
       requireConst(s)
-      inline if !constValue[Matches[s.type, ".+:.+:.+"]] then
-        error("Invalid function token")
+      inline if !constValue[Matches[s.type, ".+:.+:.+"]] then error("Invalid function token")
```

Comment:
`scalafmt` insisted on this formating

Reply:
artifact from maxColumns=100 time I think, it's fine